examples/sensor-websocket/src/components/SensorDisplay.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from "react";
+import {
+  Button,
+  Col,
+  Container,
+  FormControl,
+  InputGroup,
+  Row,
+  Spinner
+} from "react-bootstrap";
+import CredentialsModal from "./CredentialsModal";
+import SensorDataHandler from "./SensorDataHandler";
+import ApiHandler from "../apiHandler";
+
+class SensorDisplay extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      visible: false,
+      device: null,
+      astarte: null,
+      loading: false,
+      submit: false
+    };
+  }
+
+  componentDidMount() {
+    const config = localStorage.AstarteConfig;
+    if (config) {
+      const astarte = new ApiHandler(JSON.parse(config));
+      this.setState({ astarte, visible: false });
+    } else {
+      this.setState({ visible: true });
+    }
+  }
+
+  setCredentials = config => {
+    const astarte = new ApiHandler(config);
+    localStorage.AstarteConfig = JSON.stringify(config);
+    this.setState({ astarte, visible: false }, () => this.handleSubmit());
+  };
+
+  handleChange = e => {
+    e.preventDefault();
+    this.setState({ [e.target.name]: e.target.value, submit: false });
+  };
+
+  handleCredentialModal = visible => {
+    this.setState({ visible });
+    if (!visible) {
+      this.handleSubmit();
+    }
+  };
+
+  handleSubmit = () => {
+    const { device, astarte } = this.state;
+    if (device) {
+      this.setState({ loading: true });
+      astarte
+        .getDevice(device)
+        .then(response => this.setDeviceData(response))
+        .catch(error => this.handleError(error));
+    }
+  };
+
+  setDeviceData(response) {
+    this.setState({
+      device: response.id,
+      loading: false,
+      submit: true
+    });
+  }
+
+  handleError(err) {
+    if (err.response.status === 403)
+      this.setState({ visible: true, loading: false });
+    if (err.response.status === 404) {
+      this.setState({ loading: false });
+      window.confirm("Device ID Invalid");
```

Comment:
404 is "not found". Invalid request is bound to error code 400

Reply:
When we hit API with wrong Device ID it returns 404 status code

Reply:
let me know about this

Reply:
not really. take for example `AAAAAAAAAAAAAAAAAAAAAA` is a valid device ID. If your realm contains a device with such a device ID, the API will return 200 (Ok), otherwise 404 (Not found).
On the other hand, `fakeid` is not a valid ID, so the API will always return 400 (bad request)